test/integration/scheduler/predicates_test.go

```diff
@@ -0,0 +1,883 @@
+			},
+			pods: []*v1.Pod{
+				{
+					Spec: v1.PodSpec{NodeName: nodes[0].Name,
+						Containers: []v1.Container{{Name: "container", Image: framework.GetPauseImageName(cs)}},
+						Affinity: &v1.Affinity{
+							PodAntiAffinity: &v1.PodAntiAffinity{
+								RequiredDuringSchedulingIgnoredDuringExecution: []v1.PodAffinityTerm{
+									{
+										LabelSelector: &metav1.LabelSelector{
+											MatchExpressions: []metav1.LabelSelectorRequirement{
+												{
+													Key:      "service",
+													Operator: metav1.LabelSelectorOpNotIn,
+													Values:   []string{"securityscan", "value2"},
+												},
+											},
+										},
+										TopologyKey: "zone",
+									},
+								},
+							},
+						},
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:   "fake-name2",
+						Labels: podLabel},
+				},
+			},
+			node: nodes[0],
+			fits: true,
+			test: "verify that PodAntiAffinity from existing pod is respected when pod has no AntiAffinity constraints. satisfy PodAntiAffinity symmetry with the existing pod",
+		},
+		{
+			pod: &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{Name: "fake-name2"},
+				Spec: v1.PodSpec{
+					Containers:   []v1.Container{{Name: "container", Image: framework.GetPauseImageName(cs)}},
+					NodeSelector: map[string]string{"region": "r1"},
+					Affinity: &v1.Affinity{
+						PodAntiAffinity: &v1.PodAntiAffinity{
+							RequiredDuringSchedulingIgnoredDuringExecution: []v1.PodAffinityTerm{
+								{
+									LabelSelector: &metav1.LabelSelector{
+										MatchExpressions: []metav1.LabelSelectorRequirement{
+											{
+												Key:      "foo",
+												Operator: metav1.LabelSelectorOpIn,
+												Values:   []string{"abc"},
+											},
+										},
+									},
+									TopologyKey: "region",
+								},
+							},
+						},
+					},
+				},
+			},
+			pods: []*v1.Pod{
+				{Spec: v1.PodSpec{
+					Containers: []v1.Container{{Name: "container", Image: framework.GetPauseImageName(cs)}},
+					NodeName:   nodes[0].Name}, ObjectMeta: metav1.ObjectMeta{Name: "fakename", Labels: map[string]string{"foo": "abc"}}},
+			},
+			fits: false,
+			test: "nodes[0] and nodes[1] have same topologyKey and label value. Node2 has an existing pod that match the inter pod affinity rule. The pod can not be scheduled onto either of the two nodes.",
+		},
+	}
+	for _, test := range tests {
+		for _, pod := range test.pods {
+			var nsName string
+			if pod.Namespace != "" {
+				nsName = pod.Namespace
+			} else {
+				nsName = context.ns.Name
+			}
+			createdPod, err := cs.Core().Pods(nsName).Create(pod)
+			if err != nil {
+				t.Fatalf("Test Failed: error, %v, while creating pod during test: %v", err, test.test)
+			}
+			err = wait.Poll(time.Second, wait.ForeverTestTimeout, podScheduled(cs, createdPod.Namespace, createdPod.Name))
+			if err != nil {
+				t.Errorf("Test Failed: error, %v, while waiting for pod during test, %v", err, test)
+			}
```

Comment:
@bsalamat, i did thought of using that during initial patch and then reason i decided not to use "runPausePod" is because one of the test case involves pod creation failure, please refer L843. I could have used "runPausePod" for existing pods but then for the sake of consistency and clarity, thought of skipping to use it. Also using complete pod objects might be clearer than using pausePodConfig.
Thinking on above lines, i opted for not using runPausePod.